test/cri-containerd/policy_test.go

```diff
@@ -814,3 +815,85 @@ func Test_RunContainer_WithPolicy_And_SecurityPolicyEnv_Annotation(t *testing.T)
 		}
 	}
 }
+
+func Test_RunContainer_WithPolicy_And_SecurityPolicyEnv_Dropping(t *testing.T) {
+	requireFeatures(t, featureLCOW, featureLCOWIntegrity)
+	pullRequiredLCOWImages(t, []string{imageLcowK8sPause, imageLcowAlpine})
+
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// The command prints environment variables to stdout, which we can capture
+	// and validate later
+	alpineCmd := []string{"ash", "-c", "env && sleep 1"}
+
+	for _, config := range []struct {
+		name    string
+		policy  string
+		allowed bool
+	}{
+		{
+			name:    "Dropped",
+			policy:  alpineSecurityPolicy(t, "rego", true, securitypolicy.WithCommand(alpineCmd)),
+			allowed: false,
+		},
+		{
+			name:    "NotDropped",
+			policy:  alpineSecurityPolicy(t, "rego", false, securitypolicy.WithCommand(alpineCmd)),
+			allowed: false,
+		},
+	} {
+		t.Run(config.name, func(t *testing.T) {
+			sandboxRequest := sandboxRequestWithPolicy(t, config.policy)
+
+			podID := runPodSandbox(t, client, ctx, sandboxRequest)
+			defer removePodSandbox(t, client, ctx, podID)
+			defer stopPodSandbox(t, client, ctx, podID)
+
+			containerRequest := getCreateContainerRequest(
+				podID,
+				"alpine-with-policy",
+				imageLcowAlpine,
+				alpineCmd,
+				sandboxRequest.Config,
+			)
+
+			// setup logfile to capture stdout
+			logPath := filepath.Join(t.TempDir(), "log.txt")
+			containerRequest.Config.LogPath = logPath
+
+			badKV := &runtime.KeyValue{
+				Key: "INVALID_ENV", Value: "this/should/cause/an/error/",
+			}
+			containerRequest.Config.Envs = append(containerRequest.Config.Envs, badKV)
+
+			response, err := client.CreateContainer(ctx, containerRequest)
+			if err != nil {
+				if config.allowed {
```

Comment:
what is `allowed` config for? `client.CreateContainer` only creates metadata in containerd, so the actual enforcement happens during `startContainer(t, client, ctx, containerID)` request. So if you want to catch any policy errors and ignore/fail based on this config, that's when they'll surface.

Reply:
Thanks!